### PR TITLE
Improve test image creation

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Comment by mvollmer** [comment]
The actual image creation needs to be tested.  I'll tweak hubbot first to make this better.

**Comment by mvollmer** [comment]
My plan with this pull request is:

- We watch the image creation in action.  (It will create the "0" tag for cockpit and ipa.)
- The image creation will fail (fraw, ...) or the image will be broken (f22, ...)
- I will (one last time) manually install our current images with the "0" tag.
- We watch this pull request use those images and tests will hopefully pass as they used to.
- We merge this pull request.
- I remove the old untagged images.

In parallel, I make new pull request that increases the tag to "1" for f22 and fix the bugs that this uncovers in that pull request.  That will be the first real test for the new scheme.

**Comment by mvollmer** [comment]
Hmm, the tag needs to be per flavor/os combination, not just per flavor.

**Comment by mvollmer** [comment]
Image creation for fraw fails as expected (known issue in systemd, link tba). I have installed the old images also with tag "0".

**Comment by mvollmer** [comment]
Image creation for cockpit f22 succeeds as expected, but there is trouble with ipa f22.  Manually testing the new cockpit f22 image shows that there are genuine problems with newer versions of Docker and kubernetes.  I have installed the cockpit f22 images also with tag "0".

The trouble with ipa f22:

```
Traceback (most recent call last):
  File "./vm-create", line 52, in <module>
    machine.save()
  File "/home/hubbot/hubbot/b10df454bd5b766d4fb7290827203f1c8a810736_f22_x86-64/test/testvm.py", line 595, in save
    assert not self._process
AssertionError
```

**Comment by mvollmer** [comment]
Image creation for cockpit f22-atomic succeeds, but I haven't tested the resulting image.   I have installed the old cockpit f22-atomic images also with tag "0".

**Comment by mvollmer** [comment]
Second try to create ipa f22 succeeded.

**Comment by mvollmer** [comment]
r7 image creation has succeeded and tests have passed, so I leave the new image in place.

**Comment by mvollmer** [comment]
> Second try to create ipa f22 succeeded.

However, it didn't get tested, so I have installed the old image also with the "0" tag.

**Comment by mvollmer** [comment]
> My plan with this pull request is:
>
> - We watch the image creation in action.  (It will create the "0" tag for cockpit and ipa.)
> - The image creation will fail (fraw, ...) or the image will be broken (f22, ...)
> - I will (one last time) manually install our current images with the "0" tag.
> - We trigger this pull request again and watch it use those images and the tests will hopefully pass >   as they used to.

We are at this point now.  Tests have passed or failed, as usual.

> - We merge this pull request.
> - I remove the old untagged images.
> 
> In parallel, I make new pull request that increases the tag to "1" for f22 and fix the bugs that this
> uncovers in that pull request.  That will be the first real test for the new scheme.

This is #2430.

**Comment by mvollmer** [comment]
Further improvements that might come later:

- Put all files for a given flavor/os/arch/tag combination into a directory instead of keeping them together with a checksum manifest.

- Let hubbot or some other service publish images automatically.

- Let hubbot garbage collect unused images, similar to how it garbage collects result directories.

- Change vm-create so that it doesn't create missing images automatically, but needs a flag for that.  Hubbot would pass that flag, but when used locally, it wouldn't launch into a lengthy create unexpectedly when a image can't be downloaded.

File: test/fedora.install [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+supervised_virt_install $connect \
+                        --initrd-inject=$ks \
+                        --extra-args="ks=file:/$ks console=ttyS0,115200" \
+                        --name=$name \
+                        --disk "path=$out,format=qcow2" \
+                        --ram 4096 \
+                        --vcpus=1 \
+                        --os-type linux \
+                        --os-variant fedora21 \
```
Comment (dperpeet) [inline test/fedora.install]: We could run into trouble here if we expand our testing repertoire... should we add a mapping for our current choices and leave the default blank (or even revert to fedora21)?

File: test/testvm.py [hunk 2]
```diff
@@ -573,60 +577,45 @@ def unpack_base(self, modify_func=None):
...
-                os.unlink(self._image_image)
+        elif os.path.isfile(magic_base_image):
             self.message("Creating disk copy:", self._image_image)
-            shutil.copyfile(image_file, self._image_image)
-            self._image_kernel = None
-            self._image_initrd = None
-            if modify_func:
-                self.run_modify_func(modify_func)
-
+            shutil.copyfile(magic_base_image, self._image_image)
```
Comment (dperpeet) [inline test/testvm.py]: The reason we perform a full copy here is so we can download images more easily if something goes wrong. Should we add that as a comment here? If we don't perform a full copy here, downloading the prepared image via create will fail without the magic base image.

File: test/README [hunk 3]
```diff
@@ -141,41 +145,20 @@ option, a default prefix will be used.  The suffices "F0", "F1", "F2",
...
 A test machine image created by vm-create doesn't contain any Cockpit
 code in it yet.  You can build and install the currently checked out
 working copy of Cockpit like this:
 
   $ ./vm-install $(../tools/make-rpms)
 
-This will modify the test machine image in ./run, but will not modify
-the saved version in $TEST_DATA/images.
+This will modify the test machine image that is used for the next run,
+but will not modify the saved version in $TEST_DATA.  Use vm-reset to
```
Comment (dperpeet) [inline test/README]: wouldn't this be in $TEST_DATA/images?

File: test/fedora.install (hunk 1 above)
Comment (mvollmer) [inline test/fedora.install]: Hmm, we need to tell virt-install that this is Fedora, and relying on auto detection is only asking for trouble. Fedora 21 is the highest fedora version supported for the --os-variant option right now by virt-install, so... I think this is fine.

File: test/testvm.py (hunk 2 above)
Comment (mvollmer) [inline test/testvm.py]: I don't think we need to explain that.

File: test/README (hunk 3 above)
Comment (mvollmer) [inline test/README]: Fixed.

**Pull request closed without merging**